### PR TITLE
Removes React anti pattern + nicer error handling.

### PR DESCRIPTION
**Pull request by kutyepov**
fetch-gif.js used to get a reference to react component and call `setState` on it.
It's an anti pattern in react world. It's better to have `this.setState` within
a component itself. It was not very clear for students where imageURL comes from
and was a bit difficult for volunteers to explain it due students'
lack of knowledge of JS function context etc.

### CODE REVIEW

File: src/fetch-gif.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
 async function fetchGIF(apiKey, searchTerm) {
-  if (!apiKey || !searchTerm) {
-    return;
+  if (!apiKey) {
+    throw new Error('Please provide API key. You may want to refer to slides to get one or reach out to a volunteer.');
+  }
+  if(!searchTerm) {
+    throw new Error('There is no search term. You may want to define search term something like a "cat", "dog" or whatever you like.')
```
Comment (faranjessani) [inline src/fetch-gif.js]: Missing a semi colon

Comment (faranjessani) [reply src/fetch-gif.js]: Here and below

File: src/index.js [hunk 2]
```diff
@@ -34,14 +46,14 @@ class App extends React.Component {
  * Send the data down from the parent App component!
  */
 function Giphy() {
```
Comment (faranjessani) [inline src/index.js:48]: Can we remove this entirely and stick with one style of components?

**Review by faranjessani — CHANGES_REQUESTED** [review]

File: src/fetch-gif.js (hunk 1 above)
Comment (kutyepov) [reply src/fetch-gif.js]: fixed

File: src/index.js [hunk 3]
```diff
@@ -34,14 +46,14 @@ class App extends React.Component {
...
-  return <div>Where's my data? :(</div>;
+  return <div>It looks like we have successfully loaded a gif... But why is not showing up here?! :)</div>;
```
Comment (faranjessani) [inline src/index.js:49]: Does this require a semi colon?

**Review by faranjessani — APPROVED** [review]

**Pull request merged**